src/api/app/controllers/webui/staging/concerns/groups_and_users_cache.rb

```diff
@@ -0,0 +1,32 @@
+module Webui
+  module Staging
+    module Concerns
+      module GroupsAndUsersCache
+        extend ActiveSupport::Concern
+
+        private
+
+        def cache_groups_and_users
+          # as it is not expected that there are many groups (~30) we catch all of them. Otherwise use this instead:
+          # group_ids = Review.where(bs_request: BsRequest.where(staging_project: @staging_projects)).select('group_id').distinct
+          # Group.where(id: group_ids).each do |group|
+          #
+          # TODO: Refactor this code using to_h when updating to Ruby 2.6 (performance improvement)
+          Rails.cache.fetch(Group.all.cache_key) do
+            @groups_hash = {}
+            Group.find_each do |group|
+              @groups_hash[group.title] = group
+            end
+          end
+
+          # TODO: Refactor this code using to_h when updating to Ruby 2.6 (performance improvement)
+          @users_hash = {}
+          user_ids = Review.where(bs_request: BsRequest.where(staging_project: @staging_project || @staging_projects)).select('user_id').distinct
```

Comment:
you could cache this as well, see my overall comment.

Reply:
we need to merge it today, maybe we can break it in two PRs @saraycp  @ChrisBr ?

Reply:
Fine for me...

Reply:
@vpereira 

> we need to merge it today, maybe we can break it in two PRs @saraycp @ChrisBr ?

how? :confused: